### PR TITLE
Distance Dialog.

### PR DESCRIPTION
**Pull request by shadrackkibet**
@africanmathsinitiative/developers  this is ready for review.

### CODE REVIEW

File: instat/dlgStringDistance.vb [hunk 1]
```diff
@@ -116,8 +116,9 @@ Public Class dlgStringDistance
         ucrInputPatternStringDistance.SetRCode(clsStringDistFunction, bReset)
         ucrInputComboBoxMethod.SetRCode(clsStringDistFunction, bReset)
         ucrSaveStringDistance.SetRCode(clsStringDistFunction, bReset)
-        ucrPnlStringDist.SetRCode(clsCurrentFunction, bReset)
-
+        If bReset Then
+            ucrPnlStringDist.SetRCode(clsCurrentFunction, bReset)
+        End If
```
Comment (maxwellfundi) [inline instat/dlgStringDistance.vb]: Why is this being run only when on breset?

File: instat/dlgStringDistance.vb [hunk 2]
```diff
@@ -89,26 +89,21 @@ Public Class dlgStringDistance
         clsStringDistFunction = New RFunction
         clsColumnFunction = New RFunction
 
-        ucrInputPatternStringDistance.SetName("")
         ucrSelectorStringDistance.Reset()
         ucrSaveStringDistance.Reset()
-        ucrInputComboBoxMethod.Reset()
-        ucrInputPatternStringDistance.Reset()
```
Comment (maxwellfundi) [inline instat/dlgStringDistance.vb]: Are these not needed anymore or no reseting needs to be done on them? if so why?

File: instat/dlgStringDistance.vb (hunk 1 above)
Comment (shadrackkibet) [reply instat/dlgStringDistance.vb]: This is the way i found to fix reset of the panel when you reopen the dialog. I wanted when i reopen the dialog this doesn't run. Do you have a better way to deal with this?

File: instat/dlgStringDistance.vb (hunk 2 above)
Comment (shadrackkibet) [reply instat/dlgStringDistance.vb]: This are not needed anymore. We have the Rcode set for this controls and RDefault for the combobox which will reset this controls automatically.

Comment (maxwellfundi) [reply instat/dlgStringDistance.vb]: Great looks good.

**Pull request merged**